app/livechat/server/methods/changeLivechatStatus.js

```diff
@@ -1,20 +1,45 @@
 import { Meteor } from 'meteor/meteor';
 
 import { Livechat } from '../lib/Livechat';
+import { hasPermission } from '../../../authorization';
+import Users from '../../../models/server/models/Users';
 
 Meteor.methods({
-	'livechat:changeLivechatStatus'() {
-		if (!Meteor.userId()) {
+	'livechat:changeLivechatStatus'({ status, agentId = Meteor.userId() }) {
+		const uid = Meteor.userId();
+
+		if (!uid || !agentId) {
 			throw new Meteor.Error('error-not-allowed', 'Not allowed', { method: 'livechat:changeLivechatStatus' });
 		}
 
-		const user = Meteor.user();
+		const agent = Users.findOneAgentById(agentId, {
+			fields: {
+				status: 1,
+				statusLivechat: 1,
+			},
+		});
+
+		const newStatus = status || (agent.statusLivechat === 'available' ? 'not-available' : 'available');
+
+		if (newStatus === agent.statusLivechat) {
+			return;
+		}
+
+		if (!agent) {
```

Comment:
Please, move this check above [this line](https://github.com/RocketChat/Rocket.Chat/pull/18571/files#diff-65dd30e37b206dbbe498aac01e9ab994R22). Otherwise, `agent.statusLivechat` may raise an error when the `agent` is not found.